codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -1165,4 +1132,36 @@ private boolean allowModifications() {
             return pendingIterations == 0;
         }
     }
+
+    /**
+     * Implementation of {@link PropertyKey} that specifies the index position of the property.
+     */
+    class DefaultPropertyKey implements PropertyKey {
+        private final int index;
+
+        DefaultPropertyKey(int index) {
+            this.index = index;
+        }
+    }
+
+    /**
+     * A registry of all stream property keys known by this connection.
+     */
+    private class PropertyKeyRegistry {
+        final List<DefaultPropertyKey> keys = new ArrayList<DefaultPropertyKey>(4);
```

Comment:
Anything static would require synchronization, which seems like overkill.  Let's defer doing this until we have a legitimate use case.